### PR TITLE
[infra] Print additional information for build/coverage failures on GCB (#3104).

### PR DESCRIPTION
**Pull request by Dor1s**
Haven't tested yet, will do before landing.

Inspired by https://github.com/GoogleCloudPlatform/cloud-builders/issues/253, as it seems like there doesn't exist a better mechanism for this.

### CODE REVIEW

**Comment by Dor1s** [comment]
Here's a log example: https://pantheon.corp.google.com/logs/viewer?resource=build%2Fbuild_id%2Fa4507385-5b0f-4d47-87b4-a02cbadeaf00&project=oss-fuzz&minLogLevel=0&expandAll=false&timestamp=2020-01-08T22:06:26.904000000Z&customFacets=&limitCustomFacetWidth=true&dateRangeStart=2020-01-08T21:06:27.156Z&dateRangeEnd=2020-01-08T22:06:27.156Z&interval=PT1H&scrollTimestamp=2020-01-08T21:55:45.583709541Z

This works, but need to add ` && false` after the `echo`.

**Comment by Dor1s** [comment]
Here are newer examples:

https://pantheon.corp.google.com/logs/viewer?resource=build%2Fbuild_id%2F13506292-1b5f-4bbe-9c48-9657931ba593&project=oss-fuzz&minLogLevel=0&expandAll=false&timestamp=2020-01-08T22:35:31.885000000Z&customFacets=&limitCustomFacetWidth=true&dateRangeStart=2020-01-08T21:35:32.136Z&dateRangeEnd=2020-01-08T22:35:32.136Z&interval=PT1H&scrollTimestamp=2020-01-08T22:22:35.747846919Z

https://pantheon.corp.google.com/logs/viewer?resource=build%2Fbuild_id%2Febdf990a-3087-42d1-b063-96af5148581a&project=oss-fuzz&minLogLevel=0&expandAll=false&timestamp=2020-01-08T22:35:36.976000000Z&customFacets=&limitCustomFacetWidth=true&dateRangeStart=2020-01-08T21:35:37.229Z&dateRangeEnd=2020-01-08T22:35:37.229Z&interval=PT1H&scrollTimestamp=2020-01-08T22:23:19.870125122Z

**Comment by Dor1s** [comment]
Here should be a good example of coverage failures due to missing corpus backup: https://console.developers.google.com/logs/viewer?resource=build%2Fbuild_id%2Fd883b9be-3a40-4c6c-8b45-8fa608d47041&project=oss-fuzz

File: infra/gcb/build_project.py [hunk 1]
```diff
@@ -276,7 +279,11 @@ def get_build_steps(project_dir):
               {
                   'name': 'gcr.io/oss-fuzz-base/base-runner',
                   'env': env,
-                  'args': ['bash', '-c', 'test_all'],
+                  'args': [
+                      'bash',
+                      '-c',
+                      'test_all || echo "check_build %s" && false' % failure_msg
```
Comment (oliverchang) [inline infra/gcb/build_project.py]: nit: use .format() for consistency with the rest?

Comment (Dor1s) [reply infra/gcb/build_project.py]: done!

**Comment by Dor1s** [comment]
Now this is the real good log for a coverage failure: https://pantheon.corp.google.com/logs/viewer?resource=build%2Fbuild_id%2Fc9e50508-4d3c-4cc8-8a2b-ad6e1b8b220b&project=oss-fuzz&minLogLevel=0&expandAll=false&timestamp=2020-01-09T17:41:13.813000000Z&customFacets=&limitCustomFacetWidth=true&dateRangeStart=2020-01-09T16:41:14.064Z&dateRangeEnd=2020-01-09T17:41:14.064Z&interval=PT1H&scrollTimestamp=2020-01-09T17:34:57.487027292Z

Landing a testing on a few real builds to double-check that everything is ok.

**Pull request merged**